tests/Support/SupportLazyCollectionTest.php

```diff
@@ -231,4 +231,23 @@ public function testUniqueDoubleEnumeration()
 
         $this->assertSame([1, 2], $data->all());
     }
+
+    public function testTakeWithNegativeLimit()
+    {
+        $data = LazyCollection::times(10);
+
+        $this->assertSame([
+            7 => 8,
+            8 => 9,
+            9 => 10,
+        ], $data->take(-3)->all());
+
+        $this->assertSame([
+            7 => 8,
+            8 => 9,
+            9 => 10,
+        ], $data->take(-5)->take(-3)->all());
+
+        $this->assertSame($data->take(10)->all(), $data->take(-10)->all());
+    }
```

Comment:
This belongs in the regular `Collection` tests, which tests both collection classes.

It's not specific to the lazy collection.

Reply:
I'm sorry, what does that mean?

The intention of adding this test is to check if there are any flaws in my refactored behavior.

Perhaps I should have verified that Collection::take() and LazyCollection::take() with the same elements produce the same result?

Reply:
@fuwasegu have you seen [this file](https://github.com/laravel/framework/blob/10.x/tests/Support/SupportCollectionTest.php)? Every method there runs the same tests for both the `Collection` and `LazyCollection` classes. It uses [a PHPUnit data provider](https://tighten.com/insights/tidying-up-your-phpunit-tests-with-data-providers/) for this:

https://github.com/laravel/framework/blob/b6f47323da0f0052680d337fe49ffdfe12692ad6/tests/Support/SupportCollectionTest.php#L5683-L5689

I've been meaning to write a blog post about this testing method, using these tests as a showcase. I actually have a draft, but didn't get around to finishing it yet.

Reply:
I see!
Sorry, I did not know this file existed.

It is true that it is DRY and very cool to test the functionality of multiple classes that implement the same interface by efficiently generating instances of each using a data provider!

Reply:
I reverted it